telethon/network/mtproto_sender.py

```diff
@@ -341,7 +362,6 @@ def _clear_all_pending(self):
 
         # TODO For now, simply ack msg_new.answer_msg_id
         # Relevant tdesktop source code: https://goo.gl/VvpCC6
-        await self._send_acknowledge(msg_new.answer_msg_id)
```

Comment:
Hm, I thought the point was to acknowledge as soon as possible?

Reply:
[There](#diff-f2a9579ee6161ac0d9cc176fb73086c7R119) I send acknowledge. But you send there and there ([1](#diff-f2a9579ee6161ac0d9cc176fb73086c7L177) -> [2](#diff-f2a9579ee6161ac0d9cc176fb73086c7L76)) -- twice.

Reply:
Yes but acknowledging `remote_msg_id` is *not* acknowledging the `answer_msg_id`.

Reply:
Ok, yes, I see. You are right! [There](https://core.telegram.org/mtproto/service_msg_about_messages) is also said that the base `MsgDetailedInfo` does not requires acknowledge.

Probably need to check other types of messages which doesn't require acknowledges.

Reply:
So, this line should be kept removed or added back? Why does Telegram Desktop acknowledge it?:
```cpp
bool received = false;
MTPlong resMsgId = data.vanswer_msg_id; /* here */
{
    QReadLocker lock(sessionData->receivedIdsMutex());
    received = (sessionData->receivedIdsSet().lookup(resMsgId.v) != ReceivedMsgIds::State::NotFound);
}
if (received) {
    ackRequestData.push_back(resMsgId); /* here */
} else {
    ...
```